v1/topdown/save.go

```diff
@@ -365,7 +365,10 @@ func saveRequired(c *ast.Compiler, ic *inliningControl, icIgnoreInternal bool, s
 		}
 		switch node := node.(type) {
 		case *ast.Expr:
-			found = len(node.With) > 0 || ignoreExprDuringPartial(node)
+			found = len(node.With) > 0
+			if !ic.nondeterministicBuiltins { // skip evaluating non-det builtins for PE
+				found = ignoreExprDuringPartial(node)
+			}
```

Comment:
☝️ This is what it's about, really. The test is pushing the configurable through all the layers of HTTP, CLI and the Rego package.